Sources/Bow/Typeclasses/Monad.swift

```diff
@@ -1,50 +1,135 @@
 import Foundation
 
+/// A Monad provides functionality to sequence operations that are dependent from one another.
+///
+/// Instances of `Monad` must obey the following rules:
+///
+///     flatMap(pure(a), f) == f(a)
+///     flatMap(fa, pure) == fa
+///     flatMap(fa) { a in flatMap(f(a), g) } == flatMap(flatMap(fa, f), g)
+///
+/// Also, instances of `Monad` derive a default implementation for `Applicative.ap` as:
+///
+///     ap(ff, fa) == flapMap(ff, { f in map(fa, f) }
 public protocol Monad: Applicative {
+    /// Sequentially compose two computations, passing any value produced by the first as an argument to the second.
+    ///
+    /// - Parameters:
+    ///   - fa: First computation.
+    ///   - f: A function describing the second computation, which depends on the value of the first.
+    /// - Returns: Result of composing the two computations.
     static func flatMap<A, B>(_ fa: Kind<Self, A>, _ f: @escaping (A) -> Kind<Self, B>) -> Kind<Self, B>
+
+    /// Monadic tail recursion.
+    ///
+    /// - Parameters:
+    ///   - a: Initial value for the recursion.
+    ///   - f: A function describing a recursive step.
+    /// - Returns: Result of evaluating recursively the provided function with the initial value.
     static func tailRecM<A, B>(_ a: A, _ f : @escaping (A) -> Kind<Self, Either<A, B>>) -> Kind<Self, B>
 }
 
+// MARK: Related functions
+
 public extension Monad {
+    // Docs inherited from `Applicative`
     public static func ap<A, B>(_ ff: Kind<Self, (A) -> B>, _ fa: Kind<Self, A>) -> Kind<Self, B> {
         return self.flatMap(ff, { f in map(fa, f) })
     }
-    
+
+    /// Flattens a nested structure of the context implementing this instance into a single layer.
+    ///
+    /// - Parameter ffa: Value with a nested structure.
+    /// - Returns: Value with a single context structure.
     public static func flatten<A>(_ ffa: Kind<Self, Kind<Self, A>>) -> Kind<Self, A> {
         return self.flatMap(ffa, id)
     }
     
+    /// Sequentially compose two computations, discarding the value produced by the first.
+    ///
+    /// - Parameters:
+    ///   - fa: 1st computation.
+    ///   - fb: 2nd computation.
+    /// - Returns: Result of running the second computation after the first one.
     public static func followedBy<A, B>(_ fa: Kind<Self, A>, _ fb: Kind<Self, B>) -> Kind<Self, B> {
         return self.flatMap(fa, { _ in fb })
     }
-    
+
+    /// Sequentially compose a computation with a lazy computation, discarding the value produced by the first.
```

Comment:
`Eval` is not always lazy, right? We could rephrase it as:

```suggestion
    /// Sequentially compose a computation with a potentially lazy one, discarding the value produced by the first.
```